test/TestLedgerProcess.hs

```diff
@@ -0,0 +1,192 @@
+              <account-total>
+                <amount>
+                  <commodity flags="S">
+                    <symbol>SGD</symbol>
+                  </commodity>
+                  <quantity>-5</quantity>
+                </amount>
+              </account-total>
+            </account>
+          </account>
+          <account>
+            <name>Expenses</name>
+            <fullname>Expenses</fullname>
+            <account-total>
+              <amount>
+                <commodity flags="S">
+                  <symbol>SGD</symbol>
+                </commodity>
+                <quantity>5</quantity>
+              </amount>
+            </account-total>
+            <account>
+              <name>Food</name>
+              <fullname>Expenses:Food</fullname>
+              <account-amount>
+                <amount>
+                  <commodity flags="S">
+                    <symbol>SGD</symbol>
+                  </commodity>
+                  <quantity>5</quantity>
+                </amount>
+              </account-amount>
+              <account-total>
+                <amount>
+                  <commodity flags="S">
+                    <symbol>SGD</symbol>
+                  </commodity>
+                  <quantity>5</quantity>
+                </amount>
+              </account-total>
+            </account>
+          </account>
+        </account>
+      </accounts>
+      <transactions>
+        <transaction>
+          <date>2018/01/01</date>
+          <payee>food</payee>
+          <postings>
+            <posting>
+              <account>
+                <name>Expenses:Food</name>
+              </account>
+              <post-amount>
+                <amount>
+                  <commodity flags="S">
+                    <symbol>SGD</symbol>
+                  </commodity>
+                  <quantity>5</quantity>
+                </amount>
+              </post-amount>
+              <total>
+                <amount>
+                  <commodity flags="S">
+                    <symbol>SGD</symbol>
+                  </commodity>
+                  <quantity>5</quantity>
+                </amount>
+              </total>
+            </posting>
+            <posting generated="true">
+              <account>
+                <name>Assets:SGD</name>
+              </account>
+              <post-amount>
+                <amount>
+                  <commodity flags="S">
+                    <symbol>SGD</symbol>
+                  </commodity>
+                  <quantity>-5</quantity>
+                </amount>
+              </post-amount>
+              <total>
+                <amount>
+                  <commodity flags="S">
+                    <symbol>SGD</symbol>
+                  </commodity>
+                  <quantity>0</quantity>
+                </amount>
+              </total>
+            </posting>
+          </postings>
+        </transaction>
+      </transactions>
+    </ledger>
+
```

Comment:
1. Having the blank newline at the end is a bit sloppy. The code ought to be able to work even without two newlines at the end, right?

2. There's **no way** that all of that XML document is meaningful. -- the `Xml` module below already showcases what the assumptions about document structure are.

3. Missing case: what happens if the 'ledger' program isn't on the `PATH`?